### PR TITLE
feat(document_loader): Add optional selector

### PR DESCRIPTION
**Pull request by magick93**
This adds an optional selector parameter to the `load()` function. The type is then specified in `langchain/src/document_loaders/cheerio_web_base.ts`

### CODE REVIEW

File: langchain/src/document_loaders/base.ts [hunk 1]
```diff
@@ -10,7 +10,7 @@ export interface DocumentLoader {
 }
 
 export abstract class BaseDocumentLoader implements DocumentLoader {
-  abstract load(): Promise<Document[]>;
+  abstract load(selector?: unknown): Promise<Document[]>;
```
Comment (nfcampos) [inline langchain/src/document_loaders/base.ts:13]: Thanks for this, I think the selector should be moved to a constructor argument on CheerioWebBaseLoader

File: langchain/src/document_loaders/cheerio_web_base.ts [hunk 2]
```diff
@@ -47,9 +47,9 @@ export class CheerioWebBaseLoader
     );
   }
 
-  async load(): Promise<Document[]> {
+  async load(selector?: SelectorType): Promise<Document[]> {
     const $ = await this.scrape();
-    const text = $("body").text();
+    const text = $(selector).text() ?? $("body").text();
```
Comment (nfcampos) [inline langchain/src/document_loaders/cheerio_web_base.ts:52]: This should instead be `$(selector ?? "body").text();`

**Review by nfcampos — COMMENTED** [review]
two comments

File: langchain/src/document_loaders/base.ts (hunk 1 above)
Comment (arronKler) [reply langchain/src/document_loaders/base.ts:13]: I do think so, its weird to add a parameter 'selector' in the abstract BaseDocumentLoader, especially you do implementation in other tools. Its not an universal parameter

File: langchain/src/document_loaders/cheerio_web_base.ts (hunk 2 above)
Comment (magick93) [reply langchain/src/document_loaders/cheerio_web_base.ts]: Nice!

**Comment by nfcampos** [comment]
Hi @magick93 can you give me permission to push to this branch (there should be a checkbox you can tick here in the PR page) so I can fix the conflicts, and move the selector the fields object

**Comment by magick93** [comment]
@nfcampos

I couldnt find the checkbox - but merge conflicts should be resolved now.

**Pull request merged**